backend/src/bin/abacus.rs

```diff
@@ -120,3 +131,35 @@ async fn shutdown_signal() {
         _ = terminate => {},
     }
 }
+
+#[cfg(test)]
+mod test {
+    use rand::Rng;
+    use sqlx::SqlitePool;
+    use test_log::test;
+
+    use super::start_server;
+
+    pub fn random_port() -> u16 {
+        let mut rng = rand::rng();
+
+        rng.random_range(10_000..30_000)
+    }
```

Comment:
You can use port 0 to let the OS assign a random unused port, see e.g. https://github.com/tokio-rs/axum/blob/da3539cb0e5eed381361b2e688a776da77c52cd6/examples/sse/src/main.rs#L84-L96

Reply:
Fixed in f73f0af371a002f77982e34e36dab7cba9d97aa8. In addition, in e48839a3e08e6882f0ec2fea01db4890df2d4e6c, I have moved the new function and test to the library so that it can also be used in the integration test.